types/datum.go

```diff
@@ -1128,16 +1128,10 @@ func (d *Datum) convertToMysqlTimestamp(sc *stmtctx.StatementContext, target *Fi
 	}
 	switch d.k {
 	case KindMysqlTime:
-		// `select timestamp(cast("1000-01-02 23:59:59" as date)); ` casts usage will succeed.
-		// Alter datetime("1000-01-02 23:59:59") to timestamp will error.
-		if sc.InReorgAttribute {
```

Comment:
Could remove the field of `InReorgAttribute` in SC? It seems to be used only for `KindMysqlTime`.

Reply:
Good catch! Thank you for spotting it, I have removed it (basically this PR will replace https://github.com/pingcap/tidb/pull/26395 apart from the added tests).